src/kernels/kernelProvider.ts

```diff
@@ -86,18 +95,23 @@ export class KernelProvider implements IKernelProvider {
         this._onDidRestartKernel.dispose();
         this._onKernelStatusChanged.dispose();
     }
-    public getOrCreate(notebook: NotebookDocument, options: KernelOptions): IKernel {
-        const existingKernelInfo = this.kernelsByNotebook.get(notebook);
+    public getOrCreate(uri: Uri, options: KernelOptions): IKernel {
+        const existingKernelInfo = this.getInternal(uri);
+        const notebook = workspace.notebookDocuments.find((nb) =>
```

Comment:
I think this needs to use the findNotebook api because workspace.notebookDocuments doesn't include interactive notebooks.

Reply:
But maybe I'm remembering wrong.

Reply:
> workspace.notebookDocuments 

Nope, it does include the interactive window notebook as well.